src/functions-reference/higher-order_functions.Rmd

```diff
@@ -7,173 +7,148 @@ by name to the higher-order functions.
 
 ```{r results='asis', echo=FALSE}
 if (knitr::is_html_output()) {
-cat(' * <a href="functions-algebraic-solver.html">Algebraic Equation Solver</a>\n')
+cat(' * <a href="functions-algebraic-solver.html">Algebraic Equation Solvers</a>\n')
 cat(' * <a href="functions-ode-solver.html">Ordinary Differential Equation (ODE) Solvers</a>\n')
 cat(' * <a href="functions-1d-integrator.html">1D Integrator</a>\n')
 cat(' * <a href="functions-reduce.html">Reduce-Sum</a>\n')
 cat(' * <a href="functions-map.html">Map-Rect</a>\n')
 }
 ```
 
-## Algebraic equation solver {#functions-algebraic-solver}
+## Algebraic equation solvers {#functions-algebraic-solver}
 
 Stan provides two built-in algebraic equation solvers,
-respectively based on Powell's and Newton's methods.
-The Newton method constitutes a more recent addition to Stan;
+respectively based on the Newton method and the Powell "dog leg" hybrid method.
+Empirically the Newton method is found to be faster and
 its use is recommended for most problems.
-Although they look like other function applications, 
-algebraic solvers are special in two ways.
 
-First, an algebraic solver is a higher-order function, i.e. it takes
+An algebraic solver is a higher-order function, i.e. it takes
 another function as one of its arguments. Other functions in
-Stan which share this feature are the ordinary differential equation
-solvers (see section [Ordinary Differential Equation (ODE) Solvers](#functions-ode-solver)).
+Stan which share this feature are the differential equation
+solvers (see section [Ordinary Differential Equation (ODE) Solvers](#functions-ode-solver) and
+[Differential Algebraic Equation (DAE) solver](#functions-dae-solver)).
 Ordinary Stan functions do not allow functions as arguments.
 
-Second, some of the arguments of the algebraic solvers are restricted
-to data only expressions. These expressions must not contain variables
```

Comment:
While it is correct to delete this paragraph, it might be good to mention the computational advantages of marking arguments as `data` when needed in the discussion of the variadic arguments.

Reply:
Ah, I didn't realize arguments had to be explicitly marked using `data` to avoid propagating derivatives with respect to them (I guess that makes sense). Here's an additional paragraph:


There is no type restriction for the variadic arguments
and each argument can be passed as data or parameter.
However users should use parameter arguments only when nescessary,
and mark data arguments with the keyword `data`.
In the below example, the last variadic argument, $x$, is restricted to being data:

```stan
  vector algebra_system (vector y, vector theta, data vector x)
```

Distinguishing data and parameter is important for computational reasons.
Augmenting the total number of parameters increases the cost of propagating derivatives
through the solution to the algebraic equation,
and ultimately the computational cost of evaluating the gradient of the target.